scalafix/rules/src/main/scala/fix/StreamAppRules.scala

```diff
@@ -0,0 +1,51 @@
+package fix
+
+import scalafix.lint.LintSeverity
+import scalafix.v1._
+
+import scala.meta._
+
+object StreamAppRules {
+  def unapply(t: Tree)(implicit doc: SemanticDocument): Option[Patch] =
+    t match {
+      case Defn.Class(_, _, _, _, tpl) =>
+        Some(replaceStreamApp(tpl) + replaceStreamMethod(tpl))
+      case Defn.Object(_, _, tpl) =>
+        Some(replaceStreamApp(tpl) + replaceStreamMethod(tpl))
+      case exitCodeSuccessMatcher(fs2ExitCode @ Name(_)) =>
+        Some(Patch.lint(Diagnostic("StreamAppExitCode", message = "You can remove this", position = fs2ExitCode.pos, severity = LintSeverity.Warning)))
+      case i@Importee.Name(Name("StreamApp")) =>
+        Some(Patch.removeImportee(i) + addCatsEffectImports + addCatsSyntaxImport)
+      case _ =>
+        None
+    }
+
+  private[this] def replaceStreamApp(tpl: Template): Patch =
+    tpl.inits.collect{
+      case s@Init(Type.Apply(Type.Name("StreamApp"), List(Type.Name("IO"))), _, _) =>
+        Patch.replaceTree(s, "IOApp")
+    }.asPatch
+
+  private[this] def replaceStreamMethod(tpl: Template)(implicit doc: SemanticDocument): Patch =
+    tpl.stats.collect{
+      case d@Defn.Def(_, _, _, _, _, body) =>
+        val newDef = d.copy(
+          name = Term.Name("run"),
+          paramss = List(List(Term.Param(List(), Name("args"), Some(Type.Apply(Type.Name("List"), List(Type.Name("String")))), None))),
+          decltpe = Some(Type.Apply(Type.Name("IO"), List(Type.Name("ExitCode")))),
+          body = Term.Apply(Term.Select(Term.Select(Term.Select(body, Term.Name("compile")), Term.Name("drain")), Term.Name("as")), List(Term.Select(Term.Name("ExitCode"), Term.Name("Success"))))
+        )
+      Patch.replaceTree(d, newDef.toString)
+    }.asPatch
+
+  private[this] def addCatsEffectImports(implicit c: SemanticContext): Patch =
+    Patch.addGlobalImport(Symbol("cats/effect/IO.")) +
+      Patch.addGlobalImport(Symbol("cats/effect/ExitCode.")) +
+        Patch.addGlobalImport(Symbol("cats/effect/IOApp."))
+
+  private[this] def addCatsSyntaxImport(implicit c: SemanticDocument): Patch =
+    Patch.addGlobalImport(importer"cats.syntax.functor._")
```

Comment:
Is it possible to make this detect whether cats syntax is already present, and if so skip this?

Also it might be good to check for both `cats.implicits._` as well as functor syntax

Reply:
Yeah this is a refinement I want to do in a second time. For now I'm focusing on the migration itself. Though good point! I'll keep it in mind :)

Reply:
Fixed in https://github.com/functional-streams-for-scala/fs2/pull/1304/commits/ba16b95bc4df6aa87558453370f5f5ea203a6d17